writer.go

```diff
@@ -54,10 +55,27 @@ func NewWriter(w io.Writer, opts ...OptionFunc) *Writer {
 	return writer
 }
 
+// GetWriter returns a new Writer based on request param `type` that writes to w.
+// if r have the query param `type`="variable" - we set VariableLengthFields to `true` and NewlineCharacter as ""
+// else the writer defaults to fixed-length fields and use "\n" for NewlineCharacter.
+func GetWriter(w io.Writer, r *http.Request) *Writer {
+	// default writer
+	var writer = NewWriter(w)
+
+	// check for query param `type`. if "variable" then update writer
+	var fileType = r.URL.Query().Get("type")
+	if fileType == "variable" {
+		writer = NewWriter(w, VariableLengthFields(true), NewlineCharacter(""))
```

Comment:
I'm not sure about changing the newline character here. This could surprise people who were only expecting this query parameter to change fixed vs variable length.

Reply:
can separate put as a boolean query param i.e.
with no or newline=true for default("\n") as newline character
and newline=false for "" new line

Reply:
I could see exposing another query param for that. Please just make sure that omitting any one, or both, of these params will result in the same behavior that currently exists in master. That will ensure we don't accidentally introduce a breaking change to the behavior of the API.

Reply:
added tests to ensure the current behavior works as expected.
If no query param is passed - we return as expected and honor query params independently